project-docs/stumptown-writers-guide.md

```diff
@@ -0,0 +1,697 @@
+    * if `chapter_list` then this section contains links to all pages are specified using a link list or `chapter_list`, then this is a link list, specified using a directory or a chapter list.
+    if `children`, then this section itself contains sections: this enables you to nest groups of links to an arbitrary depth.
+
+So for example, here's a "related content" file for HTML:
+
+```yaml
+- title: Learn HTML
+  children:
+  - chapter_list: "/content/learn/html/Introduction_to_HTML.yaml"
+  - chapter_list: "/content/learn/html/Multimedia_and_embedding.yaml"
+- title: Guides
+  children:
+  - chapter_list: "/content/html/guides/Guides.yaml"
+- title: Reference
+  children:
+  - title: Elements
+    directory: "/content/html/reference/elements"
+  - title: Global attributes
+    directory: "/content/html/reference/attributes"
+```
+
+It consists of three top-level sections.
+
+* The first section "Learn HTML" consists of two subsections, each specified using a chapter list ("Introduction_to_HTML.yaml" and "Multimedia_and_embedding.yaml").
+
+* The second section "Guides" consists of one subsection, specified using a chapter list ("Guides.yaml").
+
+* The third section "Reference" consists of two subsections, each specified using a directory ("/content/html/reference/elements" and "/content/html/reference/attributes").
+
+The resulting structure looks like:
+
+* Learn HTML
+    * Introduction to HTML
+        * ... all Introduction to HTML chapters
+    * Multimedia and embedding
+        * ... all Multumedia and embedding chapters
+* Guides
+    * HTML guides
+        * ...all HTML guide pages
+* Reference
+    * Elements
+        * ...all HTML element pages
+    * Global attributes
+        * ...all HTML global attribute pages
+
+These related content files are referenced using the `related_content` property. This may appear either in recipes (for page types that all want the same related content collection), or in the front matter section of individual pages (for page types, like guides and landing pages, that don't all want the same related content collection).
+
+**Note: we will need to think about how to extend this grammar for more complex sidebars.**
+
+**Note: we should probably abandon this "infinite levels of sections" thing here and perhaps restrict ourselves to three levels (by not allowing children to contain children).**
+
+## Chapter lists
+
+A "chapter list" is just a way to provide a reusable list of links to pages, presented in a particular order. For example, a sidebar might want to list all the HTML guide pages in a curated order, with the most basic first, and more advanced concepts later. So the sidebar specification could provide its own list of pages to include. But an HTML landing page might also want to include the same list, in the same order. This is what chapter lists enable you to do.
+
+A chapter list is a YAML file with the following contents:
```

Comment:
It would be nice if we could have the ability to divide the chapter list up into titled sections, to be rendered like

**The basics**
- Introducing HTML
- Adding styles
- Showing pictures

**Forms and inputs**
* Adding forms to pages
* Input type guide
... etc...

Reply:
I think we should leave this until and unless we have a definite need for it.